topics/data-science/tutorials/python-math/tutorial.md

```diff
@@ -0,0 +1,361 @@
+>
+> > ### {% icon solution %} Solution
+> > ```
+> > x = 452
+> > y = x * 921 + 534
+> > # Remember to print it out!
+> > print(y)
+> > ```
+> >
+> > 416826
+> {: .solution}
+{: .question}
+
+```python
+# Test solutions here!
+# By the way, lines starting with a # are comment lines!
+# You can use that to take notes, and not affect how your code runes
+# People use it for documentation: explaining what the function does,
+# what a variable means, why they chose this or that algorithm.
+```
+
+## Functions
+
+Just like you might have done in math, we can have functions in Python. You do not need to know this now, but it's help you understand what's going on underneath. The important thing to notice is how we call our function, just like in math, we write `f(3)` to call a function named `f` with the value `3`!
+
+> > ### {% icon code-in %} Math
+> > ```
+> > # Define our function
+> > f(x) = 3 * x
+> > # Compute some value
+> > f(3) # is 9
+> > ```
+> {: .code-in}
+>
+> > ### {% icon code-out %} Python
+> > ```
+> > # Define our function
+> > def f(x):
+> >    answer = 3 * x
+> >    return answer
+> > # Compute some value
+> > f(3)
+> > ```
+> {: .code-out}
+>
+{: .code-2col}
+
+
+## Libraries
+
+A library is a collection of files (called modules) that contains functions for use by other programs. It may also contain data values (e.g., numerical constants) and other things. A library's contents are supposed to be related, but there's no way to enforce that. The Python standard library is an extensive suite of modules that comes with Python itself. Many additional libraries are available from [PyPI (the Python Package Index)](https://pypi.org/).
+
+### Libraries and modules
+
+A library is a collection of modules, but the terms are often used interchangeably, especially since many libraries only consist of a single module, so don't worry if you mix them.
+
+### A program must import a library module before using it.
+
+You can use `import` to load a library module into a program's memory, then refer to things from the `module as module_name.thing_name`. Python uses `.` to mean “part of”. For example, using `datetime`, one of the modules in the standard library:
+
+```python
+import datetime
+
+# it is currently
+datetime.datetime.now()
+```
+
+
+You can use `help` to learn about the contents of a library module. it works just like help for a function.
+
+```python
+help(datetime)
+```
+
+You can import specific items from a library module to shorten programs. You can use `from ... import ...` to load only specific items from a library module. Then refer to them directly without library name as prefix.
+
+
+```python
+from datetime import datetime
+
+datetime.now()
+```
+
+
+You can create an alias for a library module when importing it to shorten programs. Use `import ... as ...` to give a library a short alias while importing it. Then refer to items in the library using that shortened name.
+
+```python
+import datetime.datetime as dt
```

Comment:
@hexylena 
this gives an error, maybe change to 
```
from datetime import datetime as dt

dt.now()
```

Reply:
I've just stuffed this in a `tip` box, it's too irrelevant anyway, not sure why carpentries thought it was useful.